### PR TITLE
Assessment results

### PR DESCRIPTION
**Pull request by dczulada**
QRDA Cat I R 4 Templates that report results should use the Result Template, instead of the value element.

Diagnostic Study Performed (V3) - 2.16.840.1.113883.10.20.24.3.18 
Intervention Performed (V3 - 2.16.840.1.113883.10.20.24.3.32
Procedure Performed (V3) - 2.16.840.1.113883.10.20.24.3.64
Assessment Performed - 2.16.840.1.113883.10.20.24.3.144

This change is at the request of the QRDA Implementation Group.

There was one issue with the Medication Allergy template using the wrong extension version.  That is also fixed here.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1360
- [x] Internal ticket links back to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] Code coverage has not gone down and all code touched or added is covered.
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here:
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here:
 
**Cypress Reviewer:**
 
Name: @okeefm
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code
 
**Bonnie Reviewer:**
 
Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: templates/cat1/r4/_results.cat1.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<% entry.values.each do |value| %>
+  <entryRelationship typeCode="REFR">
+   <observation classCode="OBS" moodCode="EVN">
+     <!-- Conforms to C-CDA R2 Result Observation (V2) -->
+     <templateId root="2.16.840.1.113883.10.20.22.4.2" extension="2015-08-01"/>
+     <!-- Result (QRDA I R3) -->
+     <templateId root="2.16.840.1.113883.10.20.24.3.87" extension="2016-02-01"/>
+     <id root="1.3.6.1.4.1.115" extension="<%= identifier_for(value) %>"/>
```
Comment (edeyoung) [inline templates/cat1/r4/_results.cat1.erb:8]: `id root="..."` appears to be different in the QRDA r4 example.
In the examples, it's `<id root="ed971e73-c060-4258-8838-69c830254b5a" />`

**Review by edeyoung — CHANGES_REQUESTED** [review]
Please see comment re id. Not sure if it needs to be changed or not.

**Comment by edeyoung** [comment]
Original:

```
<entry>
<observation classCode="OBS" moodCode="EVN">
<!--  Assessment Performed  -->
<templateId root="2.16.840.1.113883.10.20.24.3.144" extension="2016-08-01"/>
<id root="1.3.6.1.4.1.115" extension="594855bcb84846433c0fa406"/>
<code code="72136-5" codeSystem="2.16.840.1.113883.6.1" sdtc:valueSet="2.16.840.1.113883.3.117.1.7.1.357">
<originalText>Assessment, Performed: Vte Risk Assessment</originalText>
<translation code="72136-5" codeSystem="2.16.840.1.113883.6.1"/>
</code>
<text>Assessment, Performed: Vte Risk Assessment</text>
<statusCode code="completed"/>
<effectiveTime>
<low value="20121206063000"/>
<high value="20121206064500"/>
</effectiveTime>
<value code="62482003" codeSystem="2.16.840.1.113883.6.96" xsi:type="CD" sdtc:valueSet="2.16.840.1.113883.3.117.1.7.1.400">
<originalText>Low Risk</originalText>
</value>
</observation>
</entry>
```

Modified:
```
<entry>
<observation classCode="OBS" moodCode="EVN">
<!--  Assessment Performed  -->
<templateId root="2.16.840.1.113883.10.20.24.3.144" extension="2016-08-01"/>
<id root="1.3.6.1.4.1.115" extension="594855bcb84846433c0fa406"/>
<code code="72136-5" codeSystem="2.16.840.1.113883.6.1" sdtc:valueSet="2.16.840.1.113883.3.117.1.7.1.357">
<originalText>Assessment, Performed: Vte Risk Assessment</originalText>
<translation code="72136-5" codeSystem="2.16.840.1.113883.6.1"/>
</code>
<text>Assessment, Performed: Vte Risk Assessment</text>
<statusCode code="completed"/>
<effectiveTime>
<low value="20121206063000"/>
<high value="20121206064500"/>
</effectiveTime>
<entryRelationship typeCode="REFR">
<observation classCode="OBS" moodCode="EVN">
<!--  Conforms to C-CDA R2 Result Observation (V2)  -->
<templateId root="2.16.840.1.113883.10.20.22.4.2" extension="2015-08-01"/>
<!--  Result (QRDA I R3)  -->
<templateId root="2.16.840.1.113883.10.20.24.3.87" extension="2016-02-01"/>
<id root="1.3.6.1.4.1.115" extension="04131FBB7DF43DCF70047600DC597C97"/>
<code code="72136-5" codeSystem="2.16.840.1.113883.6.1" sdtc:valueSet="2.16.840.1.113883.3.117.1.7.1.357">
<originalText>Assessment, Performed: Vte Risk Assessment</originalText>
<translation code="72136-5" codeSystem="2.16.840.1.113883.6.1"/>
</code>
<statusCode code="completed"/>
<effectiveTime>
<low value="20121206063000"/>
<high value="20121206064500"/>
</effectiveTime>
<value code="62482003" codeSystem="2.16.840.1.113883.6.96" xsi:type="CD" sdtc:valueSet="2.16.840.1.113883.3.117.1.7.1.400">
<originalText>Low Risk</originalText>
</value>
</observation>
</entryRelationship>
</observation>
</entry>
```

Looks ok to me

**Review by edeyoung — APPROVED** [review]

**Review by okeefm — APPROVED** [review]
The ID change makes sense to me--IDs should be the same if elements have the same value, different otherwise, which is why the `identifier_for` function exists.
